lib/lualink.c

```diff
@@ -513,6 +513,23 @@ static int _ii_lead( lua_State *L )
     lua_settop(L, 0);
     return 0;
 }
+static int _ii_lead_bytes( lua_State *L )
+{
+    int nargs = lua_gettop(L);
+    if( nargs != 3 ) return 0;
+    uint8_t address = luaL_checkinteger(L, 1);
+    size_t len;
+    uint8_t *data = (uint8_t *)luaL_checklstring(L, 2, &len);
```

Comment:
I guess to me the Lua string type feels natural for representing byte arrays, whereas an object I know is a hashmap of floats doesn't. For arrays this small I reckon it doesn't really matter (though I think larger messages over the bus, like for configuration data, is an interesting idea to daydream about).

How about an array -> string translation function that lets us supply both APIs:
```lua
function arr_to_bytes(arr)
  local s = ''
  for k,v in ipairs(arr) do
    s = s .. strchar(v)
  end
  return s
end

function ii.raw(addr, data, rx_len)
  return ii.raw_bytestr(addr, arr_to_bytes(data), rx_len)
end

function ii.raw_bytestr(addr, bytestr, rx_len)
  ii_lead_bytes(addr, bytestr, rx_len)
end
```

Reply:
I guess I drank the Lua koolaid somewhat bc I don't think of tables as hashmaps-of-floats (even though they obvs are!), so a table of 'int looking things' is the closest thing to a byte array. Probably also because it *looks* like a C array of ints.

So perhaps neither of us is seeing it through our audience's eyes. But then again, perhaps *we* are the primary audience for this feature!

I think I'd prefer if we keep a single interface - let's do bytestring as you initially suggested - and if anyone asks to use a table, we can offer the `arr_to_bytes` code-snippet.